ros_cross_compile/pipeline_stages.py

```diff
@@ -28,3 +32,16 @@
                                          ('custom_script', Optional[Path]),
                                          ('custom_data_dir', Optional[Path]),
                                          ('custom_setup_script', Optional[Path])])
+
+
+class PipelineStage(metaclass=ABCMeta):
+    """Interface to represent a stage of the cross compile pipeline."""
+
+    @abstractmethod
+    def __init__(self, name):
+        self._name = name
+
+    @abstractmethod
+    def __call__(self, platform: Platform, docker_client: DockerClient, ros_workspace_dir: Path,
+                 customizations: PipelineStageConfigOptions):
+        pass
```

Comment:
This should probably still raise [NotImplementedError](https://docs.python.org/3/library/exceptions.html#NotImplementedError)